### PR TITLE
feat: csrf support any, fix isSafeDomain bug

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- 支持 csrf.type === 'any'
- 修复 isSafeDomain 判断不严谨的 bug

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -3,49 +3,34 @@
...
-const allowTypeList = [ 'all', 'referer', 'ctoken' ];
 const URL = require('url').URL;
 
-exports.allowTypeList = allowTypeList;
-
-/**
- * check csrf type
- * @param {String} type the csrf type
- */
-exports.checkCsrfType = type => {
```
Comment (dead-horse) [inline lib/utils.js:15]: 这里过度抽象了..

File: lib/utils.js [hunk 2]
```diff
@@ -3,49 +3,34 @@
...
     if (rule.includes('*')) {
       return matcher.isMatch(domain, rule);
     }
     // If domain is an absolute path such as `http://...`
     // We can directly check whether it directly equals to `domain`
     // And we don't need to cope with `endWith`.
-    return domain === rule || hostname.endsWith(rule);
+    if (domain === rule) return true;
+    // ensure wwweggjs.com not match eggjs.com
+    if (!/^\./.test(rule)) rule = `.${rule}`;
```
Comment (dead-horse) [inline lib/utils.js:32]: 这里还隐藏了一个漏洞

**Review by fengmk2 — APPROVED** [review]

**Comment by fengmk2** [comment]
@dead-horse 你来发版本。

**Comment by dead-horse** [comment]
2.8.0

**Pull request merged**